manifests/proxy.pp

```diff
@@ -9,17 +9,19 @@
   $_host = $parts[0]
   $_port = $parts[1]
 
-  exec{'set rhsm proxy_hostname':
-    command => "/usr/bin/subscription-manager config --server.proxy_hostname=${_host}",
-    # Only run if the proxy hostname isn't set or doesn't match the new value.
-    unless  => "/usr/bin/subscription-manager config --list | grep -q 'proxy_hostname = ${_host}'",
-    before  => Exec['RHSM-register'],
-  }
+  if ($rhsm::enabled) {
```

Comment:
I suspect we probably do. I can work on that in a bit unless you have time.

Reply:
I addressed this, but in hindsight it doesn't really make sense that this is needed. Since this is a define type, it should never get called without a `$host_port` parameter. The `else` section is more about if the `$host_port` is empty, which should never be the case here.

Reply:
@v1peractual I'm rethinking if this really needs to a define type and rely on a profile in a control repo to call it.

Perhaps this simply be puppet logic in a `rhsm::proxy` subclass that does the lookup of to see if a proxy should be set. The following code could just be moved into this `rhsm::proxy` subclass:
```
  $httpproxyhash = lookup('yum::config_options', Hash, first, {})
  if has_key($httpproxyhash, 'proxy') {
    $httpproxy = $httpproxyhash['proxy']
    if ( $httpproxy =~ String[1] ) {
       # RHSM PROXY LOGIC HERE
    } else {
       # RHSM PROXY REMOVE LOGIC HERE
    }
  }
```

Reply:
This actually probably needs to be decoupled from looking at yum's proxy configuration.

To address EUS issues in RHEL 8 for Qualys scans, we think we need to register cluster nodes with RHSM but not let RHSM manage the repositories. If that is the case, yum will not be configured with a proxy, but RHSM may still need to be configured to proxy.